src/sinks/aws_kinesis/firehose/record.rs

```diff
@@ -47,13 +50,49 @@ impl SendRecord for KinesisFirehoseClient {
     type E = KinesisError;
 
     async fn send(&self, records: Vec<Self::T>, stream_name: String) -> Option<SdkError<Self::E>> {
+        let mut r = self.inner_send(records.clone(), stream_name.clone()).await;
+
+        for _ in 1..=3 {
+            if let Ok(resp) = &r {
+                if resp.failed_put_count().unwrap_or(0) > 0 {
+                    #[cfg(not(test))] // the wait fails during test for some reason.
+                    sleep(Duration::from_millis(100)).await;
```

Comment:
There isn't any configuration or backoff at all. Really, just an example at this point to hopefully get some eyes.